src/Runner.php

```diff
@@ -156,6 +157,41 @@ public function execute($argv, $appName = null, $appVersion = null, $output = nu
         return $this->run($argv, $output, $app, $commandFiles, $this->classLoader);
     }
 
+    /**
+     * Return an initialized application loaded with specified commands and configuration.
+     *
+     * This should ONLY be used for testing purposes. Works well in conjunction with Symfony's CommandTester.
+     *
+     * @see https://symfony.com/doc/current/console.html#testing-commands
+     *
+     * @param string|null $appName
+     *   Name of the application.
+     * @param string|null $appVersion
+     *   Version of the application.
+     * @param string|null $commandFile
+     *   Name of the specific command file that should be included with the application.
+     * @param \Robo\Config\Config|null $config
+     *   Robo configuration to be used with the application.
+     * @param \Composer\Autoload\ClassLoader|null $classLoader
+     *   Class loader to use. If PHPUnit and the application runner rely on the same autoloader, this can be left null.
+     *
+     * @return \Robo\Application
+     *   Initialized application based on passed configuration and command classes.
+     */
+    public function getAppForTesting($appName = null, $appVersion = null, $commandFile = null, $config = null, $classLoader = null)
+    {
+        $app = Robo::createDefaultApplication($appName, $appVersion);
+        $output = new NullOutput();
+        $commandFiles = $this->getRoboFileCommands($output); // $output is just used for printing error messages, it can be a throwaway stream
```

Comment:
This only needs to be called if it is going to be registered.

Reply:
Do you want to leave the responsibility of command registration out of this function altogether?

Reply:
I think a lot of tests will want to just register the default list of commands. There could be two versions, though, one that took a commandfile / commandfile list, and the other that called getRoboFileCommands and called through to the first.

Reply:
Removed call to `Runner::getRoboFileCommands` and replaced it with the following check:

If the `$commandFile` parameter is a string or an array, then invoke `Runner::registerCommandClasses`. If `$commandFile` is a string, it gets converted to an array prior to being passed to `Runner::registerCommandClasses`.

Reply:
Example usage https://bitbucket.org/lpeabody/release-manager/src/fdc3554f553ac75b7fa4fd9e434f2c8278d85e33/tests/CommandTesterTrait.php?at=feature%2Frobo-966-g1a-updates#lines-25:49

Reply:
Sure, that works; on reflection, I'm not sure that `Runner::getRoboFileCommands()` is often useful in testing, as it usually just passes through what was provided to the Runner constructor.